packages/schema/src/graph.ts

```diff
@@ -173,6 +174,31 @@ export type GraphIdentifier = string;
  */
 export type SubGraphs = Record<GraphIdentifier, GraphDescriptor>;
 
+/**
+ * Properties used for testing the foundational traversal tests
+ */
+export type TestProperties = {
```

Comment:
Yeah unless we want users to use these fields, I don't think they should be here.

BTW if we convert from types to interfaces, then it will be possible to use https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces to add fields just for our own tests.

(But I wouldn't do that conversion in this PR, I tried and it actually caused a bunch of build errors).

Reply:
I do love merged declarations and exports.

Reply:
I still see the testing stuff in the PR, but this comment was resolved? We shouldn't put our internal testing stuff in public interfaces.

Reply:
I agree. I would rather make the schema less strict, allowing extra properties -- since that's likely going to happen.

Reply:
as a stop-gap before looking at migrating to interfaces I updated it so it's no longer exported and uses the `@deprecated` flag to highlight that it should be used
https://github.com/breadboard-ai/breadboard/pull/571/files#diff-cd98c3c8ef3db059cf8ac5041c98a0e771600bd33fc580fd4c78e37c0d36336aR199-R204

Reply:
happy to mark this resolved?

Reply:
Ok. Here's some inspiration for why I think we should be careful about keeping private implementation stuff out of our public interfaces: https://www.hyrumslaw.com/

Reply:
Hah, I enjoy the XKCD for illustration. I'm certainly going to reference Hyrum's Law as it's something I've thought about but not been able to articulate well in the past.
But I feel that having the schema for the testing values and having them clearly marked as such aligns with Hyrum's Law better than having them not checked/annotated or marked as invalid?